src/System.Private.CoreLib/shared/System/Text/Utf16Utility.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Diagnostics;
+
+namespace System.Text
+{
+    internal static partial class Utf16Utility
+    {
+        /// <summary>
+        /// Returns true iff the UInt32 represents two ASCII UTF-16 characters in machine endianness.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool AllCharsInUInt32AreAscii(uint value)
+        {
+            return (value & ~0x007F007Fu) == 0;
+        }
+
+        /// <summary>
+        /// Given a UInt32 that represents two ASCII UTF-16 characters, returns the invariant
+        /// lowercase representation of those characters. Requires the input value to contain
+        /// two ASCII UTF-16 characters in machine endianness.
+        /// </summary>
+        /// <remarks>
+        /// This is a branchless implementation.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static uint ConvertAllAsciiCharsInUIntToLowercase(uint value)
+        {
+            // ASSUMPTION: Caller has validated that input value is ASCII.
+            Debug.Assert(AllCharsInUInt32AreAscii(value));
+
+            // the 0x80 bit of each word of 'lowerIndicator' will be set iff the word has value >= 'A'
+            uint lowerIndicator = value + 0x00800080u - 0x00410041u;
+
+            // the 0x80 bit of each word of 'upperIndicator' will be set iff the word has value <= 'Z'
+            uint upperIndicator = value + 0x00800080u - 0x005B005Bu;
```

Comment:
The logic is right; however, the comment is not.  It should read “… will be set iff the word has value > 'Z'”.  There are 3 more comments to fix in this file.

Reply:
Good catch! Will send a new PR to fix.

Reply:
Resolved by https://github.com/dotnet/coreclr/pull/20425.